Framework/DataHandling/src/SaveNXSPE.cpp

```diff
@@ -303,4 +349,47 @@ void SaveNXSPE::exec() {
   nxFile.closeGroup(); // Top level NXentry
 }
 
+/**
+ * Calculate fixed energy for indirect instrument. Depending on actual workspace, this may be the same energy for all
+ * detectors or array of energies for all active detectors if some energies are different.
+ *
+ * @param inputWS pointer to matrix workspace with indirect instrument attached to it.
+ *
+ * @returns vector containing analyzer energies for each detector if the energies are different or single energy if they
+ * are the same
+ */
+std::vector<double> SaveNXSPE::getIndirectEfixed(const MatrixWorkspace_sptr &inputWS) const {
```

Comment:
The function in the following file is for retrieving EFixed for an Indirect workspace. Would it be possible to consolidate the two methods so we have a single method we can re-use?
https://github.com/mantidproject/mantid/blob/5ebdf0883d509dd83992ea586db0240b21530952/qt/widgets/common/src/WorkspaceUtils.cpp#L138

Reply:
At the moment the QT widgets and Framework/Datahandling are independent. I would not want to change it, at least not in the scope of this PR. In addition, there are more versions of getEFixed in other parts of Mantid. It might make sense to sanitize this at some point, but this would be a separate issue.